app/Console/Commands/ConvertToSpeech.php

```diff
@@ -57,29 +59,65 @@ public function fire()
                     $item->status = Items::STATUS_BEING_CONVERTED;
                     $item->save();
 
-                    printf("Converting content of '%s'...", $item->url);
+                    $this->info(sprintf("Converting content of '%s'...", $item->url));
+
+                    try {
+                        $response = $this->getHttpClient()->post("v1/synthesize", array(
+                            "headers" => array(
+                                'Authorization' => 'Basic ' . base64_encode(env('WATSON_TTS_USERNAME') . ':' . env('WATSON_TTS_PASSWORD')),
+                                'Content-Type'  => 'application/json',
+                                'Accept'        => 'audio/ogg; codecs=opus'
+                            ),
+                            "body" => json_encode(
+                                array(
+                                    "text" => $item->content
+                                )
+                            )
+                        ));
+
+                    } catch (ClientException $e) {
+                        $this->error($e->getMessage());
+                        $item->status = Items::STATUS_CONVERSION_FAILED;
+                        $item->save();
+                        continue;
+                    }
 
-                    $response = $this->getHttpClient()->post("", array(
-                        "body" => array(
-                            "q" => $item->content
-                        )
-                    ));
 
                     if (substr($response->getStatusCode(), 0, 1) == "2") {
-                        $filename = sprintf(static::FILENAME_FORMAT, $item->id);
+                        $filename = sprintf(static::FILENAME_FORMAT_OGG, $item->id);
 
                         $filesystem->put(
                             $filename,
                             $response->getBody()
                         );
 
-                        printf("Done (%s).\n", $filename);
+                        // Convert ogg file to mp3.
+                        $output = '';
+                        $returnCode = -1;
+                        $origPath = storage_path() . "/app/$filename";
+                        $finalPath = storage_path() . "/app/" . sprintf(static::FILENAME_FORMAT_MP3, $item->id);
+
+                        exec("ffmpeg -i $origPath -acodec libmp3lame $finalPath 2> /dev/null", $output, $returnCode);
+
+                        if ($returnCode !== 0) {
+                            $this->error('Error converting .ogg to .mp3');
+                            $item->status = Items::STATUS_CONVERSION_FAILED;
+                            $item->save();
```

Comment:
We should probably `break;` here, so we don't output a successful message (and don't delete the `ogg` file in case we just want to reconvert it?) if the conversion to `mp3` failed.